Improve the QuestionnaireResponse $extract operation handling

qligier opened the pull request:
Allows to pass a parameter `questionnaire` in the QuestionnaireResponse/$extract operation with the Questionnaire response, as discussed in #382.

Quick notes:

1. This is useless on its own: Matchbox only supports the [StructureMap-based extraction](https://build.fhir.org/ig/HL7/sdc/extraction.html#structuremap-based-extraction), so the StructureMap should be passed as well. And if we get the StructureMap, then we don't need the Questionnaire. We should only keep it if we want to support other methods as well.
2. The biggest part of the PR is the creation of `HttpRequestWrapper`, that should help us handle operations by taking care of parsing request bodies (with the right format and FHIR version), write the right response (with the right format and FHIR version), while only working with R5 models.
3. This also generates a good OperationDefinition.

In short: a good code rewrite that will make it easier to properly implement operations, but which doesn't solve #382 yet.

qligier committed:
Improve the QuestionnaireResponse $extract operation handling

qligier committed:
Improve parsing of media type

qligier committed:
Don't pass the default engine in HttpRequestWrapper

qligier committed:
Use HttpRequestWrapper in StructureMapListProvider

qligier commented:
The default API fhirVersion should be `hapi.fhir.fhir_version`

oliveregger commented:
deprecate matchbox:
  fhir:
    context:
      fhirVersion: 5.0.0

and use hapi.fhir.version (converted to other format) instead.

oliveregger commented on matchbox-server/src/main/java/ch/ahdis/matchbox/util/http/HttpRequestWrapper.java:
```diff
@@ -0,0 +1,198 @@
+		this.request = request;
+		this.response = response;
+
+		final var defaultVersion = FhirVersionEnum.R4; // todo
```
hapi.fhir.version to be used

qligier replied:
Done!

qligier committed:
Clarify use of CliContext.fhirVersion

We deprecate `matchbox.fhir.context.fhirVersion` in favor of `hapi.fhir.fhir_version`

qligier committed:
Use HttpRequestWrapper in ConformancePackageResourceProvider

qligier committed:
Add StructureMap parameter to QuestionnaireResponse/$Extract

qligier commented:
Added parameter 'structure-map' to $extract, allowing for a StructureMap resource or a FML string.
